Add 'y' type tag for 'bytes'.  Right now, it unflattens as string

ejeffrey opened the pull request:
but is never generated unless requested explicitly.  This is the first step in supporting unicode.

ejeffrey commented:
updated the pylabrad and labrad wiki

maffoo commented on labrad/test/test_types.py:
```diff
@@ -343,5 +344,9 @@ def testEvalDatetime(self):
+        foo = T.flatten('foo bar')
+        self.assertEquals(T.unflatten(foo.bytes, 'y'), 'foo bar')
+        self.assertEquals(T.unflatten(*T.flatten('foo bar', ['y'])), 'foo bar')
 if __name__ == "__main__":
```
Add newline before `if __name__ == '__main__'` check.

maffoo commented on labrad/test/test_types.py:
```diff
@@ -343,5 +344,9 @@ def testEvalDatetime(self):
         data2 = T.evalLRData(repr(data))
         self.assertEquals(data, data2)
 
+    def testUnicodeStr(self):
```
I would suggest calling this something like `testBytesType` or something, since it doesn't have anything to do with unicode yet.

maffoo commented on labrad/test/test_types.py:
```diff
@@ -343,5 +344,9 @@ def testEvalDatetime(self):
         self.assertEquals(data, data2)
 
+    def testUnicodeStr(self):
+        foo = T.flatten('foo bar')
```
add assertion that this flattens to type 's' for now.

maffoo commented on labrad/types.py:
```diff
@@ -657,6 +657,27 @@ def __flatten__(self, s, endianness):
+    """A raw 8-bit byte string."""
+    tag = 'y'
+    isFixedWidth = False
+    def __width__(self, s, endianness):
```
add blank line before `__width__` function.

maffoo commented:
we should also add the 'y' type to the hydrant module for generating random data, so that it will get exercised in places that accept arbitrary data.

ejeffrey commented:
I fixed those, and also updated LRStr() to also flatten unicode.  I think this should be OK -- the only behavior it should change would otherwise be an error.  Previously unicode failed to flatten and required you to explicitly encode it.

maffoo commented:
LGTM

jwenner commented:
@ejeffrey, is this compatible with the old Delphi manager? If not, you should probably bump the version number and, if possible, add something to block using with the Delphi manager.

ejeffrey commented:
It won't work if you actually send 'y' tagged data, but that is not every
done automatically yet.

Evan

On Mon, Oct 5, 2015 at 11:59 AM, Jim Wenner <notifications@github.com>
wrote:

> @ejeffrey <https://github.com/ejeffrey>, is this compatible with the old
> Delphi manager? If not, you should probably bump the version number and, if
> possible, add something to block using with the Delphi manager.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/labrad/pylabrad/pull/165#issuecomment-145632162>.
>
